### ISSUE TITLE
Application stops responding upon downloading data for make chaff function

### ISSUE THREAD

**Issue by banaagravrommel** [bleachbit/bleachbit#1180]
**Describe the bug**
When user tries to download data on make chaff function the application stops responding for a while. 

**To Reproduce**
1. Open the application
2. Click BleachBit Icon on the top right
3. Select Make Chaff
4. Click download data

The error happens every time.

**Expected behavior**
The application should not stop responding upon processing

**Screenshots**
https://recordit.co/0nB2IwoC1f

**Desktop**

- OS: Windows 10
- BleachBit version 4.3.0

**Additional context**
You can either add a progress bar or spinner to improve usability as well.

**Comment by az0** [comment]
Yes, IIRC, the download happens in the same thread as the UI, so downloading can freeze the UI.

**Comment by az0** [comment]
Now, it launches the download process in a separate thread while showing a spinner. These changes are visible in this video at offsets ~4 seconds to 10 seconds.

https://github.com/user-attachments/assets/55ad2a3a-27d4-4c8a-a775-6deb9b9d212b
